.generator/src/generator/templates/model_simple.j2

```
@@ -143,10 +144,12 @@ impl<'de> Deserialize<'de> for {{ name }} {
                         },
                         {%- endfor %}
                         &_ => {
-                            {%- if model.additionalProperties is defined and model.additionalProperties != false %}
+                            {%- if additionalProperties != false %}
                             if let Ok(value) = serde_json::from_value(v.clone()) {
                                 additional_properties.insert(k, value);
                             }
+                            {% else %}
+                            _unparsed = true;
```

Better to raise an exception here instead if additionalProperties is disabled